drivers/at86rf2xx/at86rf2xx.c

```diff
@@ -32,59 +35,215 @@
 #include "at86rf2xx_registers.h"
 #include "at86rf2xx_internal.h"
 #include "at86rf2xx_netdev.h"
+#include "at86rf2xx_dev_types.h"
+#include "at86rf2xx_properties.h"
+#include "at86rf212b.h"
+#include "at86rf231.h"
+#include "at86rf233.h"
+#include "at86rf233.h"
+#include "at86rfa1.h"
+#include "at86rfr2.h"
 
 #define ENABLE_DEBUG (0)
 #include "debug.h"
 
-
-void at86rf2xx_setup(at86rf2xx_t *dev, const at86rf2xx_params_t *params)
+static void at86rf2xx_setup(at86rf2xx_t *dev)
 {
+    assert(dev->base.dev_type < AT86RF2XX_DEV_TYPE_NUM_OF);
     netdev_t *netdev = (netdev_t *)dev;
-
     netdev->driver = &at86rf2xx_driver;
     /* State to return after receiving or transmitting */
-    dev->idle_state = AT86RF2XX_STATE_TRX_OFF;
+    dev->base.idle_state = AT86RF2XX_STATE_TRX_OFF;
     /* radio state is P_ON when first powered-on */
-    dev->state = AT86RF2XX_STATE_P_ON;
-    dev->pending_tx = 0;
+    dev->base.state = AT86RF2XX_STATE_P_ON;
+    dev->base.pending_tx = 0;
+
+    switch (dev->base.dev_type) {
```

Comment:
I think I would move this "dev_type" logic to an upper layer (e.g netdev). The user of the driver will always know which the variant is (e.ga "at86rfa1_setup` function could work here)